sdk/helper/keysutil/policy.go

```diff
@@ -770,6 +780,15 @@ func (p *Policy) DeriveKey(context, salt []byte, ver int, numBytes int) ([]byte,
 	}
 }
 
+func (p *Policy) safeGetKeyEntry(ver int) (*KeyEntry, error) {
+	keyVerStr := strconv.Itoa(ver)
+	keyEntry, ok := p.Keys[keyVerStr]
+	if !ok {
+		return nil, errutil.UserError{Err: "no such key version"}
+	}
+	return &keyEntry, nil
```

Comment:
Can we go back to returning by value here?

Reply:
There was some reason it didn't work, but I'll take another look.

Reply:
Nm, works just fine.  Changed.